pkg/kubevirt/core/core.go

```diff
@@ -248,6 +248,12 @@ func (p PluginSPIImpl) CreateMachine(ctx context.Context, machineName string, pr
 		},
 	}
 
+	if providerSpec.MemoryFeatures != nil && providerSpec.MemoryFeatures.Hugepages != nil {
+		virtualMachine.Spec.Template.Spec.Domain.Memory = &kubevirtv1.Memory{
+			Hugepages: providerSpec.MemoryFeatures.Hugepages,
+		}
+	}
+
```

Comment:
Why do you need this code instead of just setting `Spec.Template.Spec.Domain.Memory` to `providerSpec.MemoryFeatures` in the struct literal above? This would of course set other memory features in addition to hugepages, but isn't this even better?

Reply:
Because I don't want to pass `Guest` option yet, first we have to talk about `limits` and `request`

Reply:
take a look at for more details https://github.com/gardener/gardener-extension-provider-kubevirt/issues/36#issuecomment-701341251

Reply:
OK, let's leave as it is for now then, it can also be changed later.